README.md

```diff
@@ -1,6 +1,65 @@
-## adaXT - Fast adaptable and extendable trees for research
+## Fast Adaptable and Extendable Trees for Research
 
-**adaXT** is a Python module for tree-based regression and classification that is fast, adaptable and extendable and aims to provide researchers a more flexible workflow when building tree-based models.
+**adaXT** is a Python module for tree-based machine-learning
+algorithms that is fast, adaptable and extendable and aims to provide
+researchers a more flexible workflow when developing tree-based
+models.
 
-### Installation
-adaXT is available at [pypi](https://pypi.org/) and can be installed by running ```pip install adaXT```.
+It is distributed under the [3-Clause BSD license](LICENSE).
+
+We encourage users and developers to report problems, request
+features, ask for help, or leave general comments.
+
+Website: [https://NiklasPfister.github.io/adaXT](https://NiklasPfister.github.io/adaXT)
+
+
+### Goals
+
+The goal of adaXT is to provide a flexible and unified code-base for
+various tree-based algorithms that strikes a balance between speed and
+ease with which the code can be adapted and extended. It intends to
+provide researchers a simple toolkit for prototyping new tree-based
+algorithms.
+
+adaXT aims to provide an intuitive user experience that is similar to
+the [scikit-learn](https://scikit-learn.org) implementations of
+decision trees both in terms model-based syntax and
+hyperparameter. Under the hood, however, adaXT stikes a different
+balance between speed and ease of adapting and extending the code.
+
+#### Adaptable and extendable
+
+At the heart of any tree-based algorithm is a decision tree that can
+be fitted on data and then used to perform some version of
+prediction. adaXT has therefore been designed with a modular decision
+tree implementation that takes four input components:
+
+- Criterion class: Used during fitting to determine splits.
+
+- LeafBuilder class: Used during fitting to specify what is saved in
+  the leaf nodes.
+
+- Splitter class: Used during fitting to perform the splits.
+
+- Prediction class: Used after fitting to make predictions.
+
+By specifying these three components a range of different tree
```

Comment:
```suggestion
By specifying these four components a range of different tree
```
I know that the splitter is the same for all, but stating four inputs, and then stating 
three components seems odd. If you want to keep it as three, perhaps add the specific names
e.g. Predict, Criteria and LeafBuilder.